TestController_IngestRangeHandler test flakes out

https://github.com/treeverse/lakeFS/runs/6490112605?check_suite_focus=true failed with the words
```
time="2022-05-18T13:35:00Z" level=info msg="initialized Auth service" func=pkg/auth.NewDBAuthService file="lakeFS/pkg/auth/service.go:226"
time="2022-05-18T13:35:00Z" level=info msg="initialize OpenAPI server" func=pkg/api.Serve file="lakeFS/pkg/api/serve.go:57"
time="2022-05-18T13:35:01Z" level=info msg="schema migrated" func="pkg/db.(*DatabaseMigrator).Migrate" file="lakeFS/pkg/db/migration.go:49" direction=up host="127.0.0.1:44441" method=POST path=/api/v1/setup_lakefs request_id=cd778902-8352-4339-ab9e-25616a63143c service_name=rest_api took=18.81398ms
--- FAIL: TestController_IngestRangeHandler (1.39s)
    --- FAIL: TestController_IngestRangeHandler/successful_ingestion_no_pagination (1.39s)
        controller_test.go:1241: 
            	Error Trace:	controller_test.go:1241
            	Error:      	Not equal: 
            	            	expected: 1000
            	            	actual  : 312
            	Test:       	TestController_IngestRangeHandler/successful_ingestion_no_pagination
FAIL
```

--- a/pkg/catalog/testutils/testutils.go
+++ b/pkg/catalog/testutils/testutils.go
@@ -155,9 +155,9 @@ func (w *FakeWalker) createEntries(count int) {
 	// For example, setting "5" here will cause the test to constantly fail.
 	// Fix Bug #3384
 	//nolint:gosec
-	rand := rand.New(rand.NewSource(6))
+	randSeed := rand.New(rand.NewSource(6))
 	for i := 0; i < count; i++ {
-		relativeKey := testutil.RandomString(rand, randomKeyLength)
+		relativeKey := testutil.RandomString(randSeed, randomKeyLength)
 		fullkey := w.uriPrefix + "/" + relativeKey
 		ents[i] = store.ObjectStoreEntry{
 			RelativeKey: relativeKey,

--- a/pkg/catalog/testutils/testutils.go
+++ b/pkg/catalog/testutils/testutils.go
@@ -154,10 +154,11 @@ func (w *FakeWalker) createEntries(count int) {
 	// Calling test functions rely on Graveler to not break on the first 1000 entries.
 	// For example, setting "5" here will cause the test to constantly fail.
 	// Fix Bug #3384
+	const seed = 6
 	//nolint:gosec
-	randSeed := rand.New(rand.NewSource(6))
+	randGen := rand.New(rand.NewSource(seed))
 	for i := 0; i < count; i++ {
-		relativeKey := testutil.RandomString(randSeed, randomKeyLength)
+		relativeKey := testutil.RandomString(randGen, randomKeyLength)
 		fullkey := w.uriPrefix + "/" + relativeKey
 		ents[i] = store.ObjectStoreEntry{
 			RelativeKey: relativeKey,

--- a/pkg/catalog/testutils/testutils.go
+++ b/pkg/catalog/testutils/testutils.go
@@ -5,14 +5,15 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"math/rand"
 	"net/url"
 	"sort"
 	"time"
 
-	"github.com/thanhpk/randstr"
 	"github.com/treeverse/lakefs/pkg/catalog"
 	"github.com/treeverse/lakefs/pkg/graveler"
 	"github.com/treeverse/lakefs/pkg/ingest/store"
+	"github.com/treeverse/lakefs/pkg/testutil"
 )
 
 type FakeValueIterator struct {
@@ -148,8 +149,14 @@ const (
 
 func (w *FakeWalker) createEntries(count int) {
 	ents := make([]store.ObjectStoreEntry, count)
+
+	// Use same sequence to overcome Graveler ability to create small ranges.
+	// Calling test functions rely on Graveler to not break on the first 1000 entries.
+	// For example, setting "5" here will cause the test to constantly fail.
+	// Fix Bug #3384
+	rand := rand.New(rand.NewSource(6))
 	for i := 0; i < count; i++ {
-		relativeKey := randstr.Base64(randomKeyLength)
+		relativeKey := testutil.RandomString(rand, randomKeyLength)
 		fullkey := w.uriPrefix + "/" + relativeKey
 		ents[i] = store.ObjectStoreEntry{
 			RelativeKey: relativeKey,

--- a/pkg/catalog/testutils/testutils.go
+++ b/pkg/catalog/testutils/testutils.go
@@ -154,6 +154,7 @@ func (w *FakeWalker) createEntries(count int) {
 	// Calling test functions rely on Graveler to not break on the first 1000 entries.
 	// For example, setting "5" here will cause the test to constantly fail.
 	// Fix Bug #3384
+	//nolint:gosec
 	rand := rand.New(rand.NewSource(6))
 	for i := 0; i < count; i++ {
 		relativeKey := testutil.RandomString(rand, randomKeyLength)

